### PR TITLE
DOC: add `printoptions` as a context manager to `set_printoptions`

### PR DESCRIPTION
**Pull request by espdev**
In addition to #14426 we add the reference to `printoptions` context manager in 'Notes' section in `set_printoptions` function for better documentation.

### CODE REVIEW

**Comment by seberg** [comment]
If you check https://numpydoc.readthedocs.io/en/latest/format.html#sections the `See Also` section can contain text, and already lists this, so that would be the better thing to expand.

**Comment by espdev** [comment]
@seberg Yes, I added it to 'See Also' in previous PR, but @mattip proposed to add a slight addition to 'Notes' section. This PR does it.

https://github.com/numpy/numpy/pull/14426#issuecomment-528730913

**Comment by seberg** [comment]
Sure, I will leave it to Matti. I probably would have a slight preference to expand see also. We can add to notes as well though, but then maybe phrase it as a recommendation to prefer the other?

File: numpy/core/arrayprint.py [hunk 1]
```diff
@@ -200,6 +200,8 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     -----
     `formatter` is always reset with a call to `set_printoptions`.
 
+    `printoptions` context manager can be used to set the values temporarily.
+
```
Comment (mattip) [inline numpy/core/arrayprint.py]: This just cries out for an example, it seems like a natural continuation of "To put back the default options ...". I think "Use `printoptions` as a context manager to ..." is clearer, no?

Comment (espdev) [reply numpy/core/arrayprint.py]: > I think "Use printoptions as a context manager to ..." is clearer, no?

Probably yes, I agree that it is better.
Do we need to move this to `See Also` from `Notes`?

Comment (mattip) [reply numpy/core/arrayprint.py]: I would prefer it be in the `Examples` section with some example code.

Comment (espdev) [reply numpy/core/arrayprint.py]: Yes, but some examples already exist in `printoptions` documentation. It seems to me the reference and slight clarification in `set_printoptions` will be enough.

Comment (mattip) [reply numpy/core/arrayprint.py]: none of the examples use `printoptions` as a context manager.

Comment (espdev) [reply numpy/core/arrayprint.py]: huh, here I see examples with `printoptions` context manager.
https://numpy.org/devdocs/reference/generated/numpy.printoptions.html#numpy.printoptions

Perhaps, these examples are not detailed enough?

Comment (mattip) [reply numpy/core/arrayprint.py]: Could you copy one to the docstring of `set_printoptions`, even in place of the "To put back the default options" one? Then the text would become "To temporarily override options, use `printoptions` as a context manager" which is much nicer than the example that exists there

Comment (espdev) [reply numpy/core/arrayprint.py]: Got it. It looks good to me.

File: numpy/core/arrayprint.py [hunk 2]
```diff
@@ -236,9 +238,16 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
...
+    >>> np.set_printoptions(edgeitems=3, infstr='inf',
     ... linewidth=75, nanstr='nan', precision=8,
     ... suppress=False, threshold=1000, formatter=None)
+
+    Also to temporarily override options, use `printoptions` as a context manager:
+
+    >>> with np.printoptions(precision=2, suppress=True, threshold=5):
+    ...     np.linspace(0, 10, 10)
+    array([ 0.    1.11  2.22 ...  7.78  8.89 10.  ])
+
```
Comment (mattip) [inline numpy/core/arrayprint.py]: Almost correct, commas are missing which makes the doctest fail. Copy-paste the correct string to hopefully make the test pass:
```
Expected:
    array([ 0.    1.11  2.22 ...  7.78  8.89 10.  ])
Got:
    array([ 0.  ,  1.11,  2.22, ...,  7.78,  8.89, 10.  ])
```

Comment (espdev) [reply numpy/core/arrayprint.py]: Okay, fixed. :)

**Comment by mattip** [comment]
Thanks @espdev

**Comment by espdev** [comment]
@mattip thanks for merging!

**Pull request merged**